llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -10037,10 +10081,10 @@ void llvm::findValuesAffectedByCondition(
         Worklist.push_back(B);
       }
     } else if (match(V, m_ICmp(Pred, m_Value(A), m_Value(B)))) {
-      AddCmpOperands(A, B);
-
       bool HasRHSC = match(B, m_ConstantInt());
       if (ICmpInst::isEquality(Pred)) {
+        AddAffected(A);
+        AddAffected(B);
```

Comment:
Compile-time impact is large: http://llvm-compile-time-tracker.com/compare.php?from=f8202e1b5c4f38be95c42ab945b4f69ab7501ffa&to=6041815eace62ac98d5e1ba48b24a3dcfecde111&stat=instructions%3Au

Reply:
Is the impact here mainly from adding more affected values or from using them in isKnownNonEqual?

Something we could do is add a bitfield for what the affected value is for (e.g. knownbits, etc), so we can quickly skip ones that are not relevant.

Reply:
It is mainly from adding more affected values.

Reply:
> Something we could do is add a bitfield for what the affected value is for (e.g. knownbits, etc), so we can quickly skip ones that are not relevant.

Are you working on this?

Reply:
No, I'm not working on this.

Reply:
I have implemented a prototype (see https://github.com/llvm/llvm-project/pull/118493). But the compile-time impact looks worse: http://llvm-compile-time-tracker.com/compare.php?from=8f3cf204bb91ce09fb3651b8a7e770de7e603b2f&to=338d7fc5b72e20f1606a28f740d5a0068c95d94b&stat=instructions%3Au